pykafka/connection.py

```diff
@@ -109,11 +109,15 @@ def request(self, request):
 
     def response(self):
         """Wait for a response from the broker"""
-        size = self._socket.recv(4)
-        if len(size) == 0:
-            # Happens when broker has shut down
-            self.disconnect()
-            raise SocketDisconnectedError
+        size = b""  # Size => int32
+        while len(size) != 4:
```

Comment:
This `4` and the `4` on the next line should be put into a variable like `expected_len` or similar, since this code breaks if they're not the same number.